keyboards/ploopyco/trackball_thumb/trackball_thumb.c

```diff
@@ -0,0 +1,291 @@
+uint16_t lastMidClick      = 0;      // Stops scrollwheel from being read if it was pressed
+uint8_t  OptLowPin         = OPT_ENC1;
+bool     debug_encoder     = false;
+bool     is_drag_scroll    = false;
+
+__attribute__((weak)) void process_wheel_user(report_mouse_t* mouse_report, int16_t h, int16_t v) {
+    mouse_report->h = h;
+    mouse_report->v = v;
+}
+
+__attribute__((weak)) void process_wheel(report_mouse_t* mouse_report) {
+    // TODO: Replace this with interrupt driven code,  polling is S L O W
+    // Lovingly ripped from the Ploopy Source
+
+    // If the mouse wheel was just released, do not scroll.
+    if (timer_elapsed(lastMidClick) < SCROLL_BUTT_DEBOUNCE) {
+        return;
+    }
+
+    // Limit the number of scrolls per unit time.
+    if (timer_elapsed(lastScroll) < OPT_DEBOUNCE) {
+        return;
+    }
+
+    // Don't scroll if the middle button is depressed.
+    if (is_scroll_clicked) {
+#ifndef IGNORE_SCROLL_CLICK
+        return;
+#endif
+    }
+
+    lastScroll  = timer_read();
+    uint16_t p1 = adc_read(OPT_ENC1_MUX);
+    uint16_t p2 = adc_read(OPT_ENC2_MUX);
+    if (debug_encoder) dprintf("OPT1: %d, OPT2: %d\n", p1, p2);
+
+    int dir = opt_encoder_handler(p1, p2);
+
+    if (dir == 0) return;
+    process_wheel_user(mouse_report, mouse_report->h, (int)(mouse_report->v + (dir * OPT_SCALE)));
+}
+
+__attribute__((weak)) void process_mouse_user(report_mouse_t* mouse_report, int16_t x, int16_t y) {
+    mouse_report->x = x;
+    mouse_report->y = y;
+}
+
+__attribute__((weak)) void process_mouse(report_mouse_t* mouse_report) {
+    report_pmw_t data = pmw_read_burst();
+    if (data.isOnSurface && data.isMotion) {
+        // Reset timer if stopped moving
+        if (!data.isMotion) {
+            if (MotionStart != 0) MotionStart = 0;
+            return;
+        }
+
+        // Set timer if new motion
+        if ((MotionStart == 0) && data.isMotion) {
+            if (debug_mouse) dprintf("Starting motion.\n");
+            MotionStart = timer_read();
+        }
+
+        if (debug_mouse) {
+            dprintf("Delt] d: %d t: %u\n", abs(data.dx) + abs(data.dy), MotionStart);
+        }
+        if (debug_mouse) {
+            dprintf("Pre ] X: %d, Y: %d\n", data.dx, data.dy);
+        }
+#if defined(PROFILE_LINEAR)
+        float scale = float(timer_elaspsed(MotionStart)) / 1000.0;
+        data.dx *= scale;
+        data.dy *= scale;
+#elif defined(PROFILE_INVERSE)
+        // TODO
+#else
+        // no post processing
+#endif
+        // apply multiplier
+        // data.dx *= mouse_multiplier;
+        // data.dy *= mouse_multiplier;
+
+        // Wrap to HID size
+        data.dx = constrain(data.dx, -127, 127);
+        data.dy = constrain(data.dy, -127, 127);
+        if (debug_mouse) dprintf("Cons] X: %d, Y: %d\n", data.dx, data.dy);
+        // dprintf("Elapsed:%u, X: %f Y: %\n", i, pgm_read_byte(firmware_data+i));
+
+        process_mouse_user(mouse_report, data.dx, -data.dy);
+    }
+}
```

Comment:
This is mostly handled in the core, now.

```suggestion
report_mouse_t pointing_device_task_kb(report_mouse_t mouse_report) {
    process_wheel();

    if (is_drag_scroll) {
        mouse_report.h = mouse_report.x;
#ifdef PLOOPY_DRAGSCROLL_INVERT
        // Invert vertical scroll direction
        mouse_report.v = -mouse_report.y;
#else
        mouse_report.v = mouse_report.y;
#endif
        mouse_report.x = 0;
        mouse_report.y = 0;
    }

    return pointing_device_task_user(mouse_report);
}
```